### ISSUE TITLE
[bug] GPU not getting attached to pipeline pod via KFP v2

### ISSUE THREAD

**Issue by pankajr141** [kubeflow/pipelines#11928]
i am not able to attach GPU to a pipeline pod on KFP V2, however when running on KFP V1 it is working seamlessly.
Can you please let me know what i am doing wrong, or am i missing anything

### Environment
*  kubeflow version:
<pre>
kubeflow 1.9
</pre>

*  KFP version (Kubeflow Pipelines):
<pre>2.3.0</pre> 
*  KFP SDK version:
<pre>
kfp==2.12.1
kfp-kubernetes==1.4.0
kfp-pipeline-spec==0.6.0
kfp-server-api==2.4.0
</pre>
### Steps to reproduce
Run below code

<pre>
import kfp
from kfp import dsl, compiler, kubernetes
from kfp.dsl import component
from kfp.client import Client
from datetime import datetime

@dsl.component(
    base_image="pytorch_image:v3",
)
def task1() -> str:
    import torch
    import multiprocessing
    
    print("cuda present: ", torch.cuda.is_available())
    print(f"Num CPU: {multiprocessing.cpu_count()}")
    return "Executed on GPU"

# from kfp.dsl import container_component, ContainerSpec
# from kubernetes.client.models import V1ResourceRequirements

@dsl.pipeline
def my_pipeline():
    task = task1()
    task = task.set_accelerator_type('nvidia.com/gpu')
    task = task.set_accelerator_limit(1)
    task = task.set_caching_options(False)
    task = task.set_cpu_limit("2")

pipeline_func = my_pipeline

run_name = f"conditional-{datetime.now().strftime('%Y-%m-%d-%H:%M:%S')}"
pipeline_filename = 'conditional_flow_pipeline.yaml'
print(f"pipeline_filename: {pipeline_filename}")

compiler.Compiler().compile(pipeline_func, pipeline_filename)
endpoint = "https://172.10.123.132:31380"

client = Client(host=endpoint, verify_ssl=False)
run_details = client.create_run_from_pipeline_package(
    pipeline_filename,
    run_name=run_name,
    arguments={},
    experiment_name="username"
)
</pre>

### Expected result
Pod should be able to get attached to GPU and i should get below message inside pod
<pre>
cuda present:  True
</pre>

### Labels

---

Impacted by this bug? Give it a 👍.
